smarts/core/tests/test_multiple_traffic_data.py

```diff
@@ -114,17 +129,17 @@ def test_mutiple_traffic_data(create_scenario):
 
     traffic_history_provider = TrafficHistoryProvider()
 
-    use_first_history = True
-    for scenario in scenarios:
-        if use_first_history:
-            assert scenario.traffic_history is traffic_history_1
-        else:
-            assert scenario.traffic_history is traffic_history_2
+    for scenario, traffic_history in zip(
+        scenarios, [traffic_history_1, traffic_history_2]
+    ):
+        for key in scenario.traffic_history_service.traffic_history:
+            assert (
+                scenario.traffic_history_service.traffic_history[key]
+                == traffic_history[key]
+            )
 
         traffic_history_provider.setup(scenario)
         assert (
-            traffic_history_provider._current_traffic_history
-            == scenario.traffic_history
+            traffic_history_provider._traffic_history_service
+            == scenario.traffic_history_service
         )
```

Comment:
Are the iterations of the `for` loop independent, or can be made independent, such that they can be replaced with `map` or other constructs to improve performance and speed?